src/util.cpp

```diff
@@ -723,18 +723,18 @@ void RenameThread(const char* name)
 
 void SetupEnvironment()
 {
-#ifndef WIN32
-    try
-    {
-#if BOOST_FILESYSTEM_VERSION == 3
-            boost::filesystem::path::codecvt(); // Raises runtime error if current locale is invalid
```

Comment:
Yes, it's actually `std::locale("")` what raises the exception, which is called from within `path::codecvt()`.